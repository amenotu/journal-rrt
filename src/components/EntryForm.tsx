import * as React from 'react';
import { Form, TextArea } from 'semantic-ui-react'

export interface Props {
    onSubmitEntry: (date: string, text: string) => void;
}

export interface State {
    date?: string;
    text?: string;
}

class EntryForm extends React.Component<Props, State> {

    state = { date: '', text: '' }

    handleDateChange = (e: React.ChangeEvent<HTMLInputElement>) => this.setState({ date: e.target.value })

    handleTextChange = (e: React.ChangeEvent<HTMLInputElement>) => this.setState({ text: e.target.value })
  
    handleSubmit = () => {
        const { date, text } = this.state
        const { onSubmitEntry } = this.props;
        console.log('current state:', this.state);
        console.log( date, ' ', text);
  
        this.setState({ date: date, text: text })
        onSubmitEntry(date, text);
    }

    render() {
        const { date, text } = this.state;

        return (
            <div>
                <p>Fill out the form to add an entry!</p>
                <Form onSubmit={ this.handleSubmit }>
                    <Form.Group>
                        <Form.Field onChange={ this.handleDateChange } name='date' value={ date }>
                        <label>Date</label>
                        <input type='date'></input>
                        </Form.Field>
                        <Form.Field 
                            control= {TextArea} 
                            label='Entry' 
                            placeholder='Start typing your entry here...'
                            name='text'
                            value={ text }
                            onChange={ this.handleTextChange }
                              required/>
                        <Form.Button content='Submit' />
                    </Form.Group>
                </Form>
            </div>
        )
    }
}

export default EntryForm;
