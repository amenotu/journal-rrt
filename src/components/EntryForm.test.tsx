import * as React from 'react';
import { configure } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import EntryForm from './EntryForm';

configure({adapter: new EnzymeAdapter()});

test('EntryForm renders', () => {
    const entryform = shallow(<EntryForm onSubmitEntry={() => {}}/>);
    
    expect(entryform).toMatchSnapshot();
  });