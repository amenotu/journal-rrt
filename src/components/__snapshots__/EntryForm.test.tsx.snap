// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryForm renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EntryForm
    onSubmitEntry={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Container>
          <p>
            Fill out the form to add an entry!
          </p>
          <Form
            as="form"
            onSubmit={[Function]}
          >
            <FormGroup>
              <FormField
                control={[Function]}
                label="Date"
                name="date"
                onChange={[Function]}
                placeholder="Enter the date here..."
                required={true}
                value=""
              />
              <FormField
                control={[Function]}
                label="Entry"
                name="text"
                onChange={[Function]}
                placeholder="Start typing your entry here..."
                required={true}
                value=""
              />
              <FormButton
                as={[Function]}
                content="Submit"
                control={[Function]}
              />
            </FormGroup>
          </Form>
        </Container>,
        <Divider />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p>
              Fill out the form to add an entry!
            </p>,
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormGroup>
                <FormField
                  control={[Function]}
                  label="Date"
                  name="date"
                  onChange={[Function]}
                  placeholder="Enter the date here..."
                  required={true}
                  value=""
                />
                <FormField
                  control={[Function]}
                  label="Entry"
                  name="text"
                  onChange={[Function]}
                  placeholder="Start typing your entry here..."
                  required={true}
                  value=""
                />
                <FormButton
                  as={[Function]}
                  content="Submit"
                  control={[Function]}
                />
              </FormGroup>
            </Form>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fill out the form to add an entry!",
            },
            "ref": null,
            "rendered": "Fill out the form to add an entry!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "form",
              "children": <FormGroup>
                <FormField
                  control={[Function]}
                  label="Date"
                  name="date"
                  onChange={[Function]}
                  placeholder="Enter the date here..."
                  required={true}
                  value=""
                />
                <FormField
                  control={[Function]}
                  label="Entry"
                  name="text"
                  onChange={[Function]}
                  placeholder="Start typing your entry here..."
                  required={true}
                  value=""
                />
                <FormButton
                  as={[Function]}
                  content="Submit"
                  control={[Function]}
                />
              </FormGroup>,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <FormField
                    control={[Function]}
                    label="Date"
                    name="date"
                    onChange={[Function]}
                    placeholder="Enter the date here..."
                    required={true}
                    value=""
                  />,
                  <FormField
                    control={[Function]}
                    label="Entry"
                    name="text"
                    onChange={[Function]}
                    placeholder="Start typing your entry here..."
                    required={true}
                    value=""
                  />,
                  <FormButton
                    as={[Function]}
                    content="Submit"
                    control={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "control": [Function],
                    "label": "Date",
                    "name": "date",
                    "onChange": [Function],
                    "placeholder": "Enter the date here...",
                    "required": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "control": [Function],
                    "label": "Entry",
                    "name": "text",
                    "onChange": [Function],
                    "placeholder": "Start typing your entry here...",
                    "required": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "content": "Submit",
                    "control": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Container>
            <p>
              Fill out the form to add an entry!
            </p>
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormGroup>
                <FormField
                  control={[Function]}
                  label="Date"
                  name="date"
                  onChange={[Function]}
                  placeholder="Enter the date here..."
                  required={true}
                  value=""
                />
                <FormField
                  control={[Function]}
                  label="Entry"
                  name="text"
                  onChange={[Function]}
                  placeholder="Start typing your entry here..."
                  required={true}
                  value=""
                />
                <FormButton
                  as={[Function]}
                  content="Submit"
                  control={[Function]}
                />
              </FormGroup>
            </Form>
          </Container>,
          <Divider />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Fill out the form to add an entry!
              </p>,
              <Form
                as="form"
                onSubmit={[Function]}
              >
                <FormGroup>
                  <FormField
                    control={[Function]}
                    label="Date"
                    name="date"
                    onChange={[Function]}
                    placeholder="Enter the date here..."
                    required={true}
                    value=""
                  />
                  <FormField
                    control={[Function]}
                    label="Entry"
                    name="text"
                    onChange={[Function]}
                    placeholder="Start typing your entry here..."
                    required={true}
                    value=""
                  />
                  <FormButton
                    as={[Function]}
                    content="Submit"
                    control={[Function]}
                  />
                </FormGroup>
              </Form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fill out the form to add an entry!",
              },
              "ref": null,
              "rendered": "Fill out the form to add an entry!",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "form",
                "children": <FormGroup>
                  <FormField
                    control={[Function]}
                    label="Date"
                    name="date"
                    onChange={[Function]}
                    placeholder="Enter the date here..."
                    required={true}
                    value=""
                  />
                  <FormField
                    control={[Function]}
                    label="Entry"
                    name="text"
                    onChange={[Function]}
                    placeholder="Start typing your entry here..."
                    required={true}
                    value=""
                  />
                  <FormButton
                    as={[Function]}
                    content="Submit"
                    control={[Function]}
                  />
                </FormGroup>,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <FormField
                      control={[Function]}
                      label="Date"
                      name="date"
                      onChange={[Function]}
                      placeholder="Enter the date here..."
                      required={true}
                      value=""
                    />,
                    <FormField
                      control={[Function]}
                      label="Entry"
                      name="text"
                      onChange={[Function]}
                      placeholder="Start typing your entry here..."
                      required={true}
                      value=""
                    />,
                    <FormButton
                      as={[Function]}
                      content="Submit"
                      control={[Function]}
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "control": [Function],
                      "label": "Date",
                      "name": "date",
                      "onChange": [Function],
                      "placeholder": "Enter the date here...",
                      "required": true,
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "control": [Function],
                      "label": "Entry",
                      "name": "text",
                      "onChange": [Function],
                      "placeholder": "Start typing your entry here...",
                      "required": true,
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "content": "Submit",
                      "control": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
